src/platform/telemetry/index.ts

```diff
@@ -264,22 +276,30 @@ type TypedMethodDescriptor<T> = (
     descriptor: TypedPropertyDescriptor<T>
 ) => TypedPropertyDescriptor<T> | void;
 const timesSeenThisEventWithSameProperties = new Set<string>();
+export type PickTypeNumberProps<T, Value> = {
+    [P in keyof T as T[P] extends Value ? P : never]: T[P];
+};
+export type PickPropertiesOnly<T> = {
+    [P in keyof T as T[P] extends TelemetryEventInfo<infer R>
+        ? keyof PickType<R, number> extends never
+            ? never
+            : P
+        : never]: T[P];
+};
+
 /**
  * Decorates a method, sending a telemetry event with the given properties.
  * @param eventName The event name to send.
  * @param properties Properties to send with the event; must be valid for the event.
- * @param captureDuration True if the method's execution duration should be captured.
- * @param failureEventName If the decorated method returns a Promise and fails, send this event instead of eventName.
- * @param lazyProperties A static function on the decorated class which returns extra properties to add to the event.
- * This can be used to provide properties which are only known at runtime (after the decorator has executed).
  */
 // eslint-disable-next-line @typescript-eslint/no-explicit-any,
-export function captureTelemetry<This, P extends IEventNamePropertyMapping, E extends keyof P>(
+export function capturePerfTelemetry<This, P extends IEventNamePropertyMapping, E extends keyof PickPropertiesOnly<P>>(
```

Comment:
Previously the decorator `captureTelemetry` would automatically send a measure named `duration`, that was the default behaviour, looks like we've used this in a number of places when we've not requried the measures at all.

Hence now created two distinct functions to be specific, one to send telemetry with the perf measures `capturePerfTelemetry` and one to just capture usage of a function named `captureUsageTelemetry`

Reply:
Also these are also now strongly typed, you cannot use `capturePerfTelemetry` without having properly defined the telemetry to support a `duration` property

Reply:
Makes sense. And we can review what is using capturePerfTelemetry as you noted.